app/src/main/java/fr/free/nrw/commons/category/CategoryImagesListFragment.java

```diff
@@ -269,6 +270,35 @@ private void handleSuccess(List<Media> collection) {
         progressBar.setVisibility(GONE);
         isLoading = false;
         statusTextView.setVisibility(GONE);
+        for (Media m : collection) {
+            replaceTitlesWithCaptions(m.getDisplayTitle(), mediaSize++);
```

Comment:
I believe you can fix the problem below by replacing `replaceTitlesWithCaptions(m.getDisplayTitle(), mediaSize++);` with `replaceTitlesWithCaptions(m.get(), mediaSize++);` and performing this API call instead:
https://commons.wikimedia.org/w/api.php?action=wbgetentities&props=labels&format=json&languagefallback=1&languages=en&sites=commonswiki&ids=M80618155
Notice that this query uses `ids=` instead of `titles=`.
That removes the problem of having to hardcode `".jpg"`

Reply:
@nicolas-raoul I tried the following API. But the issue here for fetching the media we are using this API 
```
https://commons.wikimedia.org/w/api.php?action=query&format=json&formatversion=2" + //Basic parameters
            "&generator=categorymembers&gcmtype=file&gcmsort=timestamp&gcmdir=desc" + //Category parameters
            "&prop=imageinfo&iiprop=url|extmetadata&iiurlwidth=640" + //Media property parameters"&iiextmetadatafilter=DateTime|Categories|GPSLatitude|GPSLongitude|ImageDescription|DateTimeOriginal" +
            "|Artist|LicenseShortName|LicenseUrl
```

which doesn't returns the Id we need to call the above API